src/parse.rs

```diff
@@ -65,6 +64,33 @@ pub fn to_postfix(tokens: Vec<Token>) -> Result<Vec<Token>, CalcError> {
     Ok(postfixed)
 }
 
+fn walk_until_comma_or_lparen(
+    op_stack: &mut Vec<Token>,
+    postfixed: &mut Vec<Token>,
+    token: Token
+) -> Result<(), CalcError> {
+    let mut push_until_paren: bool = false;
+    while let Some(token) = op_stack.pop() {
+        if op_stack.last().map_or(true, |x| matches!(x, Token::Function(_)))
+                && token == Token::Comma
+                || token == Token::LParen {
+            push_until_paren = true;
+            break;
+        }
+        postfixed.push(token)
+    }
+
+    if matches!(token, Token::Comma) {
+        op_stack.push(token);
+    }
```

Comment:
Won't this allow two commas side by side?

Reply:
Indeed, expressions like `nroot(2,,,4)` are also valid currently, will fix that, good catch!